programs/bpf_loader/src/syscalls/mod.rs

```diff
@@ -3465,7 +3519,33 @@ mod tests {
             clean_epochschedule.warmup = src_epochschedule.warmup;
             clean_epochschedule.first_normal_epoch = src_epochschedule.first_normal_epoch;
             clean_epochschedule.first_normal_slot = src_epochschedule.first_normal_slot;
-            assert!(are_bytes_equal(&got_epochschedule, &clean_epochschedule));
+            assert!(are_bytes_equal(
+                &got_epochschedule_obj,
+                &clean_epochschedule
+            ));
+
+            let result = SyscallGetSysvar::rust(
+                &mut invoke_context,
+                epochschedule_id_va,
+                got_epochschedule_buf_va,
+                0,
+                EpochSchedule::size_of() as u64,
+                0,
+                &mut memory_mapping,
+            );
+            result.unwrap();
+
+            // clone is to zero the alignment padding
+            let epochschedule_from_buf =
+                bincode::deserialize::<EpochSchedule>(&got_epochschedule_buf)
+                    .unwrap()
+                    .clone();
```

Comment:
the test is doing the same thing as this for the same reason (the test doesnt go through the sysvar-specific syscalls, its purpose is to ensure getting those sysvars through the new syscall functions identically)
```
    // this clone looks unecessary, but it exists to zero out trailing alignment bytes
    // it is unclear whether this should ever matter
    // but there are tests using MemoryMapping that expect to see this
    // we preserve the previous behavior out of an abundance of caution
    *var = sysvar?.clone();
```

if we dont clone, the object works perfectly normal. the object equality assert would pass
```
assert_eq!(epochschedule_from_buf, src_epochschedule);
```

but byte equality would fail (`are_bytes_equal` uses raw pointers to step through the full buffer)
```
            assert!(are_bytes_equal(
                &epochschedule_from_buf,
                &clean_epochschedule
            ));
```

as noted in the comment, i dont think this should ever matter. in fact because the runtime doesnt care about the contents of the out buffer and just wants the object, it should never matter. im just being (perhaps excessively) paranoid about the fact we cant feature gate this so it has to behave identically